### ISSUE TITLE
Unable to convert a `Dictionary<string, T>`: System.Text.Json.JsonException: The JSON value could not be converted to Definitions+Reflection.

### ISSUE THREAD

**Issue by JBBianchi** [hez2010/TypedocConverter#18]
## Description ##
An error occurs when trying to convert a pseudo-dictionary from TypeScript to C#.

## Sample ##
Input
```typescript
export interface SomeKindOfDictionary {
	[prop: string]: any;
}
```

typedoc
```json
{
	"id": 0,
	"name": "@jsonforms/core",
	"kind": 0,
	"flags": {},
	"originalName": "",
	"children": [
		{
			"id": 1,
			"name": "\"packages/core/test\"",
			"kind": 1,
			"kindString": "Module",
			"flags": {
				"isExported": true
			},
			"originalName": "C:/Dev/Sandbox/jsonforms/packages/core/test.ts",
			"children": [
				{
					"id": 2,
					"name": "SomeKindOfDictionary",
					"kind": 256,
					"kindString": "Interface",
					"flags": {
						"isExported": true
					},
					"indexSignature": [
						{
							"id": 3,
							"name": "__index",
							"kind": 8192,
							"kindString": "Index signature",
							"flags": {
								"isExported": true
							},
							"parameters": [
								{
									"id": 4,
									"name": "prop",
									"kind": 32768,
									"kindString": "Parameter",
									"flags": {
										"isExported": true
									},
									"type": {
										"type": "intrinsic",
										"name": "string"
									}
								}
							],
							"type": {
								"type": "intrinsic",
								"name": "any"
							}
						}
					],
					"sources": [
						{
							"fileName": "packages/core/test.ts",
							"line": 1,
							"character": 37
						}
					]
				}
			],
			"groups": [
				{
					"title": "Interfaces",
					"kind": 256,
					"children": [
						2
					]
				}
			],
			"sources": [
				{
					"fileName": "packages/core/test.ts",
					"line": 1,
					"character": 0
				}
			]
		}
	],
	"groups": [
		{
			"title": "Modules",
			"kind": 1,
			"children": [
				1
			]
		}
	]
}
```

## Error ##
```
Unhandled exception. System.Text.Json.JsonException: The JSON value could not be converted to Definitions+Reflection. Path: $.children[0].children[0].indexSignature | LineNumber: 25 | BytePositionInLine: 24.
   at System.Text.Json.ThrowHelper.ThrowJsonException_DeserializeUnableToConvertValue(Type propertyType)
   at System.Text.Json.Serialization.Converters.ObjectWithParameterizedConstructorConverter`1.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.Converters.FSharpOptionConverter`2.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, TOption& value)
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.JsonConverter`1.TryReadAsObject(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state, Object& value)
   at System.Text.Json.Serialization.Converters.LargeObjectWithParameterizedConstructorConverter`1.ReadAndCacheConstructorArgument(ReadStack& state, Utf8JsonReader& reader, JsonParameterInfo jsonParameterInfo)
   at System.Text.Json.Serialization.Converters.ObjectWithParameterizedConstructorConverter`1.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.JsonCollectionConverter`2.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, TCollection& value)
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.Converters.FSharpOptionConverter`2.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, TOption& value)
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.JsonConverter`1.TryReadAsObject(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state, Object& value)
   at System.Text.Json.Serialization.Converters.LargeObjectWithParameterizedConstructorConverter`1.ReadAndCacheConstructorArgument(ReadStack& state, Utf8JsonReader& reader, JsonParameterInfo jsonParameterInfo)
   at System.Text.Json.Serialization.Converters.ObjectWithParameterizedConstructorConverter`1.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.JsonCollectionConverter`2.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, TCollection& value)
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.Converters.FSharpOptionConverter`2.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, TOption& value)
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.JsonConverter`1.TryReadAsObject(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state, Object& value)
   at System.Text.Json.Serialization.Converters.LargeObjectWithParameterizedConstructorConverter`1.ReadAndCacheConstructorArgument(ReadStack& state, Utf8JsonReader& reader, JsonParameterInfo jsonParameterInfo)
   at System.Text.Json.Serialization.Converters.ObjectWithParameterizedConstructorConverter`1.OnTryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.JsonConverter`1.TryRead(Utf8JsonReader& reader, Type typeToConvert, JsonSerializerOptions options, ReadStack& state, T& value)
   at System.Text.Json.Serialization.JsonConverter`1.ReadCore(Utf8JsonReader& reader, JsonSerializerOptions options, ReadStack& state)
   at System.Text.Json.JsonSerializer.ReadFromSpan[TValue](ReadOnlySpan`1 utf8Json, JsonTypeInfo jsonTypeInfo, Nullable`1 actualByteCount)
   at System.Text.Json.JsonSerializer.ReadFromSpan[TValue](ReadOnlySpan`1 json, JsonTypeInfo jsonTypeInfo)
   at System.Text.Json.JsonSerializer.Deserialize[TValue](String json, JsonSerializerOptions options)
   at Program.main$cont@83(JsonSerializerOptions jsonOptions, Config config, Unit unitVar)
   at Program.main(String[] argv)
```

**Comment by hez2010** [comment]
Which version of typedoc are you using?

**Comment by JBBianchi** [comment]
@hez2010 I think it's `0.19.2`

**Comment by hez2010** [comment]
There was a breaking change between typedoc 0.19 and 0.20. Could you try typedoc 0.20 and see if it works?

**Comment by JBBianchi** [comment]
That works, thanks a lot for you support @hez2010 !
Please, keep on the good work 🙏 Looking forward for 4.0 !

**Comment by JBBianchi** [comment]
I spoke a little too fast, there is still a little problem I just noticed.

For instance, this pieces is converted properly:
```typescript
type Properties = { [property: string]: JsonSchema4 };
```
src: https://github.com/eclipsesource/jsonforms/blob/master/packages/core/src/generators/schema.ts#L31

as
```csharp
System.Collections.Generic.Dictionary<string, JsonSchema4>? Properties { get; set; }
```

But this piece isn't:
```typescript
export interface ReferenceSchemaMap {
  [ref: string]: JsonSchema;
}
```
src: https://github.com/eclipsesource/jsonforms/blob/master/packages/core/src/util/resolvers.ts#L34-36

as
```csharp
    /// <summary>
    /// Map for storing refs and the respective schemas they are pointing to.
    /// </summary>
    interface ReferenceSchemaMap
    {
        /// <summary>
        /// Map for storing refs and the respective schemas they are pointing to.
        /// </summary>
        public JsonSchema this[string] { get => throw new System.NotImplementedException(); set => throw new System.NotImplementedException(); }
    }
```

I guess it should be something like below ?
```csharp
    /// <summary>
    /// Map for storing refs and the respective schemas they are pointing to.
    /// </summary>
    interface ReferenceSchemaMap
        : System.Collections.Generic.IDictionary<string, JsonSchema>
    {
    }
```

**Comment by hez2010** [comment]
I don't think converting indexer to `IDictionary<,>` is correct. But in your case it seems like a bug where TypedocConverter doesn't play well with the reserved keyword as a parameter name for indexer.

The correct codegen should be:
```csharp
    /// <summary>
    /// Map for storing refs and the respective schemas they are pointing to.
    /// </summary>
    interface ReferenceSchemaMap
    {
        /// <summary>
        /// Map for storing refs and the respective schemas they are pointing to.
        /// </summary>
        public JsonSchema this[string ref0] { get => throw new System.NotImplementedException(); set => throw new System.NotImplementedException(); }
    }
```

Will submit a fix soon.

**Comment by hez2010** [comment]
Merged with #21.
